### PR TITLE
[skip ci] Add doctest for LR schedulers

### PR DESCRIPTION
**Pull request by sdesrozis**
Addresses #2266

Description: Add doctest for `ParamGroupScheduler`, `LRScheduler`, and `create_lr_scheduler_with_warmup` LR Schedulers

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/handlers/param_scheduler.py [hunk 1]
```diff
@@ -788,20 +788,43 @@ class LRScheduler(ParamScheduler):
...
+            scheduler = LRScheduler(torch_lr_scheduler)
+
+            # In this example, we assume to have installed PyTorch>=1.1.0
+            # (with new `torch.optim.lr_scheduler` behaviour) and
+            # we attach scheduler to Events.ITERATION_COMPLETED
+            # instead of Events.ITERATION_STARTED to make sure to use
+            # the first lr value from the optimizer, otherwise it is will be skipped:
+            default_trainer.add_event_handler(Events.ITERATION_STARTED, scheduler)
+
+            @default_trainer.on(Events.ITERATION_COMPLETED)
```
Comment (vfdev-5) [inline ignite/handlers/param_scheduler.py:812]: @sdesrozis you should print on ITERATION_STARTED and before applying lr scheduler

Comment (sdesrozis) [reply ignite/handlers/param_scheduler.py:812]: It means the comment above in the doc is not relevant ?

Comment (vfdev-5) [reply ignite/handlers/param_scheduler.py:812]: It is relevant on how to apply the scheduler, but my comment is about when to print the values.

File: ignite/handlers/param_scheduler.py [hunk 2]
```diff
@@ -916,21 +939,40 @@ def create_lr_scheduler_with_warmup(
...
                                                         warmup_start_value=0.0,
                                                         warmup_end_value=0.1,
-                                                        warmup_duration=10,
-                                                        output_simulated_values=lr_values)
-            lr_values = np.array(lr_values)
-            # Plot simulated values
-            plt.plot(lr_values[:, 0], lr_values[:, 1], label="learning rate")
+                                                        warmup_duration=3)
+
+            default_trainer.add_event_handler(Events.ITERATION_COMPLETED, scheduler)
```
[the remark this answers is not in the document]
Comment (sdesrozis) [reply ignite/handlers/param_scheduler.py:958]: Since the default lr is 0.0, it doesn't change the doctest results, but I agree that I should update this.

[the remark this answers is not in the document]
Comment (vfdev-5) [reply ignite/handlers/param_scheduler.py:958]: Well, it is not 0.0 but 0.1, check above. I was also thinking about that and apparently you have changed that.

File: ignite/handlers/param_scheduler.py [hunk 3]
```diff
@@ -788,20 +788,43 @@ class LRScheduler(ParamScheduler):
...
-        scheduler = LRScheduler(step_scheduler)
+        .. testoutput::
 
-        # In this example, we assume to have installed PyTorch>=1.1.0
-        # (with new `torch.optim.lr_scheduler` behaviour) and
-        # we attach scheduler to Events.ITERATION_COMPLETED
-        # instead of Events.ITERATION_STARTED to make sure to use
-        # the first lr value from the optimizer, otherwise it is will be skipped:
-        trainer.add_event_handler(Events.ITERATION_COMPLETED, scheduler)
+            0.1
```
Comment (vfdev-5) [inline ignite/handlers/param_scheduler.py:820]: @sdesrozis you can see that here you are missing one value for 0.1. You are printing 2 times 0.1 and others 3 times

File: ignite/handlers/param_scheduler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sdesrozis) [reply ignite/handlers/param_scheduler.py:958]: I removed my comment because it wasnt correct, sorry.
